### ISSUE TITLE
zoom setting does not recover from zooming out

### ISSUE THREAD

**Issue by gdt** [mvglasow/satstat#142]
In the map tab, the view is widened to fit a low-accuracy location report, as is typical with cell-based location.  That's arguably very sensible.   However, the map does not return to the user-commanded zoom when the low-accuracy report is replaced by one that is more accurate.  (I realize this will get trickier if and when more providers are shown.)

Please comment with either ENOPATCH or "not a bug and I wouldn't merge such a change"...

**Comment by mvglasow** [comment]
This is “sort of” by design: There isn’t really a user-commanded zoom, other than the last zoom explicitly set by the user. When the map is in “auto-follow” mode, the map rectangle shown and the zoom are controlled automatically. The map will pan and zoom as needed to keep everything in view, i.e. all locations with their confidence intervals. It will not only zoom out if the accuracy decreases (with the circle shown becoming to big for the current map rectangle), but also if the distances between locations reported by multiple providers are too big to show at the current zoom level. (Which can happen if one provider reports bogus data, which SatStat is supposed to help diagnose.)

I originally chose to only zoom out, never in, so that the zoom level would eventually stabilize. While I am myself occasionally annoyed by the map view suddenly zooming out, I haven’t come up with a better user experience—restoring the last user-selected zoom level could easily result in the zoom level flipping back and forth.

Bottom line: I agree this is not necessarily optimal, but we first need to get an idea of what good UX would look like.

**Comment by gdt** [comment]
My notion (without reading the code yet) is to have a variable user_requested_zoom, and set that when the user pushes the + or - button.  Then, the map would try to be at that zoom, but zoom out if necessary (exactly as now).   But when the location gets better, it would return.   Given your oscillation comments, I'd say there should be some hysteresis time, where zooming in only happens if the lower zoom level would have been acceptable for the entire hold time.   Perhaps 10s is good (or 5s, or 30s) -- but probably this is best fixed vs an option, after we try a few values.

I see your point about stabilize, but for me it stabilizes at a very low zoom, typically the 1km scale, and while I get a sense of where I am, I can't see the relationship of GPS to network location results.  So I want to push buttons, which doesn't fit with driving.

With returning to last-user-set zoom, it will keep changing, but I think that's ok because I want  to see the 200m view to understand details, unless that doesn't make sense.    Someone who wants the 1km or 2km view can hit -, and then it will stay where they put it.   I suppose there could be a preference, but I wonder if the property of staying at a zoom that wasn't requested is really desired, vs the weaker property of not rapidly changing.
